doc/source/user_guide/groupby.rst

```diff
@@ -1052,18 +1065,21 @@ The ``nlargest`` and ``nsmallest`` methods work on ``Series`` style groupbys:
 Flexible ``apply``
 ------------------
 
-Some operations on the grouped data might not fit into either the aggregate or
-transform categories. Or, you may simply want GroupBy to infer how to combine
-the results. For these, use the ``apply`` function, which can be substituted
-for both ``aggregate`` and ``transform`` in many standard use cases. However,
-``apply`` can handle some exceptional use cases.
+Some operations on the grouped data might not fit into the aggregation,
+transformation, or filtration categories. For these, you can use the ``apply``
+function.
+
+.. warning::
+
+   ``apply`` has to try to infer from the result whether it should act as a reducer,
+   transformer, *or* filter, depending on exactly what is passed to it. Thus the
+   grouped column(s) may be included in the output as well as set the indices. While
```

Comment:
may be included in the output as well as set _as_ the indices

This sounds better to me, but best to double check, otherwise lgtm

Reply:
Indeed - thanks. I also changed it from "as well as" to "or" since we don't do both.

Reply:
Actually - looking at this again I think we want to say "may be included in the output or not"